digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/ProjectLinkDAO.scala

```diff
@@ -294,11 +294,12 @@ class ProjectLinkDAO {
       val roadAddressRoadPart = r.nextLongOption()
       val roadwayNumber = r.nextLong()
       val calibrationPointsSource = CalibrationPointSource.apply(r.nextIntOption().getOrElse(99))
+      val projectRoadwayNumber = r.nextLong()
 
 
       ProjectLink(projectLinkId, roadNumber, roadPartNumber, trackCode, discontinuityType, startAddrM, endAddrM, originalStartAddrMValue, originalEndAddrMValue, startDate, endDate,
         modifiedBy, linkId, startMValue, endMValue, sideCode, CalibrationPointsUtils.toProjectLinkCalibrationPointsWithSourceInfo(calibrationPoints, calibrationPointsSource), OracleDatabase.loadJGeometryToGeometry(geom), projectId,
-        status, roadType, source, length, roadwayId, linearLocationId, ely, reversed, connectedLinkId, geometryTimeStamp, roadwayNumber, Some(roadName),
+        status, roadType, source, length, roadwayId, linearLocationId, ely, reversed, connectedLinkId, geometryTimeStamp, if (roadwayNumber != 0) roadwayNumber else projectRoadwayNumber, Some(roadName),
```

Comment:
Is this check correct here: "if (roadwayNumber != 0)"? Should it be "if (roadwayNumber != NewIdValue)"?